GVFS/GVFS.Common/OrgInfoApiClient.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Class that handles communication with a server that contains version information.
+    /// </summary>
+    public class OrgInfoApiClient
+    {
+        private HttpClient client;
+        private string baseUrl;
+
+        public OrgInfoApiClient(HttpClient client, string baseUrl)
+        {
+            this.client = client;
+            this.baseUrl = baseUrl;
+        }
+
+        private string VersionUrl
+        {
+            get
+            {
+                return this.baseUrl + "/api/vfsforgitversion";
+            }
+        }
+
+        public Version QueryNewestVersion(string orgName, string platform, string ring)
+        {
+            Dictionary<string, string> queryParams = new Dictionary<string, string>()
+            {
+                { "Organization", orgName },
+                { "Platform", platform },
+                { "Ring", ring },
+            };
+
+            string responseString = this.client.GetStringAsync(this.ConstructRequest(this.VersionUrl, queryParams)).GetAwaiter().GetResult();
+            VersionResponse versionResponse = VersionResponse.FromJsonString(responseString);
+
+            return new Version(versionResponse.Version);
+        }
+
+        private string ConstructRequest(string baseUrl, Dictionary<string, string> queryParams)

```

Comment:
Since the `queryParams` are all hard coded it don't seem like we need this method and just use a `string.Format` or `$""`?

Reply:
Yeah - as is, the params are all hard coded / unchanging. There are also some optional parameters that have been discussed, that are not included in this PR.